### ISSUE TITLE
[Backward compatibility] Customer section load with one section causes an error

### ISSUE THREAD

**Issue by zhartaunik** [magento/magento2#28154]
### Preconditions (*)
1. Magento 2.4-develop
2. It cannot be reproduced on clean Magento. An issue occurred only with incompatible code.

### Steps to reproduce (*)
Open any page and take any action which force Magento to clean **one section** (In my case it was adding product to shopping cart from PDP)
Source code should contain sth like 
`customerData.reload('cart'); `
instead of 
`customerData.reload(['cart']);
`
### Expected result (*)

The console log must be clear.

### Actual result (*)

We faced with an issue in console:
_customer-data.js:87 Uncaught Error: [object Object]
    at Object.<anonymous> (customer-data.js:87)
    at fire (jquery.js:3238)
    at Object.fireWith [as rejectWith] (jquery.js:3368)
    at done (jquery.js:9848)
    at XMLHttpRequest.callback (jquery.js:10317)_

![Screenshot from 2020-09-16 14-42-31](https://user-images.githubusercontent.com/51679138/93344027-d8a90000-f839-11ea-97d5-f9d8faa3aa34.png)

### What actually happens:
www/magento/vendor/magento/module-customer/view/frontend/web/js/section-config.js:61
Previous code:
```
        /**
         * @param {*} allSections
         * @return {*}
         */
        filterClientSideSections: function (allSections) {
            if (Array.isArray(allSections)) {
                return _.difference(allSections, clientSideSections);
            }

            return allSections;
        },
```
Magento 2.3.5-p1 code:
```
        /**
         * Filters the list of given sections to the ones defined as client side.
         * @param {Array} allSections - List of sections to check.
         * @return {Array} - List of filtered sections.
         */
        filterClientSideSections: function (allSections) {
            return _.difference(_.toArray(allSections), clientSideSections);
        },
```

When we pass into this function one element (string) JS parses as an array:
```
_.toArray(allSections)
(9) ["a", "u", "t", "h", "_", "d", "a", "t", "a"]
```
And of course cannot find such sections.

**Upd 1.**
Initially, your method accepted anything and we passed a string with a section name. Now your are accepting arrays only, but there no such verification in the method.
I suppose for backward compatibility you should return back isArray verification.

**Comment by kpitn** [comment]
I confirm the bug, is anyone upgrade to 2.3.5 ?

When cart section reload url is : 
http://xxx/customer/section/load/?sections=c%2Cu%2Cs%2Ct%2Co%2Cm%2Ce%2Cr&_=1590406308206

(decode = http://xxxx/customer/section/load/?sections=c,u,s,t,o,m,e,r&_=1590406308206

**Comment by jonaschen623** [comment]
confirm bug, with version 2.3.5-p1

http://***/customer/section/load/?sections=w%2Ci%2Cs%2Ch%2Cl%2Ci%2Cs%2Ct&_=1594592363309

decode = http://***/customer/section/load/?sections=wishlist&_=1594592363309

**Comment by complexthings** [comment]
Fixed in PR: https://github.com/magento/magento2/pull/29081

**Comment by complexthings** [comment]
This happens when you get bad data in `mage-cache-storage-section-invalidation` in localStorage, usually triggered by a wildcard reload, which tries to load `*` as a section.  I have a PR that kinda of addresses this: https://github.com/magento/magento2/pull/29081

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-37811` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by gabrieldagama** [comment]
Hi @zhartaunik. Thank you for your report. 
The issue has been fixed in magento/magento2#29081 by @complexthings in 2.4-develop branch
Related commit(s):
 - [4f10b5db85a2a3387af81978e616ed0d21a77c4b](https://github.com/magento/magento2/commit/4f10b5db85a2a3387af81978e616ed0d21a77c4b)
 - [1505a36a8b21c0660e1cf519b7bd703e581789c1](https://github.com/magento/magento2/commit/1505a36a8b21c0660e1cf519b7bd703e581789c1)
 - [ff99146565f414f23a4105b3f8926d31fe4a39e8](https://github.com/magento/magento2/commit/ff99146565f414f23a4105b3f8926d31fe4a39e8)
 - [8047497701d5a332ad692844f12af11f344d9854](https://github.com/magento/magento2/commit/8047497701d5a332ad692844f12af11f344d9854)
 - [dd862a21c29de46c362a050638b277c7deda34c9](https://github.com/magento/magento2/commit/dd862a21c29de46c362a050638b277c7deda34c9)
 - [cdfc6814958dc07fe47c41725b6a0715fce76309](https://github.com/magento/magento2/commit/cdfc6814958dc07fe47c41725b6a0715fce76309)
 - [c2e254c868fec04e6d356c25c6b963ea8e02518a](https://github.com/magento/magento2/commit/c2e254c868fec04e6d356c25c6b963ea8e02518a)
 - [c4900734f8608cbadd51bb1032c2a8bfad2d0870](https://github.com/magento/magento2/commit/c4900734f8608cbadd51bb1032c2a8bfad2d0870)
 - [176013fd10f724e2a33998b02c93294ec1772a40](https://github.com/magento/magento2/commit/176013fd10f724e2a33998b02c93294ec1772a40)
 - [2922c1c293e41fa1d056d868c816b6f0f2e5c160](https://github.com/magento/magento2/commit/2922c1c293e41fa1d056d868c816b6f0f2e5c160)
 - [f7da23e991e79cb4430032bdd4170ffe9440771f](https://github.com/magento/magento2/commit/f7da23e991e79cb4430032bdd4170ffe9440771f)
 - [3b397e34ae8e05fbf08b01395bd733f08cf9bb21](https://github.com/magento/magento2/commit/3b397e34ae8e05fbf08b01395bd733f08cf9bb21)
 - [6f9a2743df524fe5cd60c71eaf7db58ca828bc54](https://github.com/magento/magento2/commit/6f9a2743df524fe5cd60c71eaf7db58ca828bc54)
 - [5de71f6075d312d78dfe2509b412646665919d8f](https://github.com/magento/magento2/commit/5de71f6075d312d78dfe2509b412646665919d8f)
 - [fd4a2c7f1759d03f54faa228b66efc7af51dd5a2](https://github.com/magento/magento2/commit/fd4a2c7f1759d03f54faa228b66efc7af51dd5a2)
 - [fae1c5c48b59d2f6b46ab47deb9e1548ec248453](https://github.com/magento/magento2/commit/fae1c5c48b59d2f6b46ab47deb9e1548ec248453)

The fix will be available with the upcoming 2.4.2 release.

**Comment by knsalim** [comment]
@kpitn  @jonaschen623 


http://***/customer/section/load/?sections=w%2Ci%2Cs%2Ch%2Cl%2Ci%2Cs%2Ct&_=1594592363309

decode = http://***/customer/section/load/?sections=wishlist&_=1594592363309

for this issues i just removed the comma(under join) in customer-data.js the issues is fixed for me 


before change 
 **sections: sectionNames.join(',')**

after change 
 **sections: sectionNames.join('')**
